pandas/core/generic.py

```diff
@@ -9344,7 +9344,7 @@ def _align_series(
             if is_series:
                 left = self._reindex_indexer(join_index, lidx, copy)
             elif lidx is None or join_index is None:
-                left = self.copy() if copy else self
+                left = self.copy(deep=copy) if copy or copy is None else self
```

Comment:
Does this mean there are still cases where `self` is being returned? (if you manually specify `copy=False`)

Reply:
```
In [1]: s1 = pd.Series([1, 2, 3])

In [2]: s2 = pd.Series([4, 5, 6])

In [3]: s11, s22 = s1.align(s2, copy=False)

In [4]: s11 is s1
Out[4]: True

In [5]: s11.iloc[0] = 10

In [6]: s1
Out[6]: 
0    10
1     2
2     3
dtype: int64
```

Of course, this is also because `copy` is already an existing keyword, so we don't just do `self = self.copy(deep=None)` under the hood as in other methods that in the past always returned a copy. 
But for other methods that already have a copy keyword, I _think_ for now we used `copy=False` to also mean shallow/lazy copy like `copy=None` with CoW enabled. 
(it's also something we still need to discuss in general what to do with those `copy` keywords)

Reply:
> But for other methods that already have a copy keyword, I _think_ for now we used `copy=False` to also mean shallow/lazy copy like `copy=None` with CoW enabled.

Actually, that's not fully correct. It seems this depends on case by case depending on how it is implemented, and we didn't consistently cover this case. Will open a dedicated issue for this topic:  https://github.com/pandas-dev/pandas/issues/50535